src/metabase/server/middleware/security.clj

```diff
@@ -186,13 +186,14 @@
 
 (defn access-control-headers
   "Returns headers for CORS requests"
-  [origin]
+  [origin uri]
   (merge
    (when
-    (approved-origin? origin (embedding-app-origin-sdk))
+    (or
+     (approved-origin? origin (embedding-app-origin-sdk))
+     (req.util/session-properties-endpoint? {:uri uri}))
```

Comment:
I wanted to move this inside the `session` endpoint itself, as opposed to having it in `access-control-headers`, but using `merge {:headers ...}` doesn't seem to work in defendpoint. If anyone has some ideas on this, that would be great.